src/components/GAddressLookup/component.js

```diff
@@ -0,0 +1,271 @@
+
+    if (highlightMatch) {
+        suggestion.structured_formatting.main_text_matched_substrings.forEach(match => {
+            const matchedTerm = suggestion.structured_formatting.main_text.slice(
+                match.offset,
+                match.length,
+            );
+            formattedLabel = suggestion.structured_formatting.main_text.replace(
+                matchedTerm,
+                `{b}${matchedTerm}{/b}`,
+            );
+
+            formattedLabel = formattedLabel.replace(/{/g, '<').replace(/}/g, '>');
+
+            formattedLabel = React.createElement('span', {
+                dangerouslySetInnerHTML: { __html: formattedLabel },
+            });
+        });
+    } else {
+        formattedLabel = suggestion.description;
+    }
+
+    return {
+        label: formattedLabel,
+        description: suggestion.structured_formatting.secondary_text,
+        icon,
+    };
+}
+
+class PlacesLookupComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.lookupId = uniqueId('gaddrlookup-input');
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSearch = this.handleSearch.bind(this);
+
+        this.state = {
+            initialized: false,
+            isSearching: false,
+            searchValue: '',
+            places: [],
+            suggestions: [],
+            userValue: props.value,
+            selectedPlace: null,
+        };
+    }
+
+    componentDidUpdate({ isScriptLoaded: prevIsScriptLoaded }) {
+        const { isScriptLoaded, isScriptLoadSucceed } = this.props;
+        if (!prevIsScriptLoaded && isScriptLoaded && isScriptLoadSucceed) {
+            this.initComponent();
+        }
+    }
+
+    getContainerClassName() {
+        const { className } = this.props;
+        return classnames('rainbow-google-address-lookup_container', className);
+    }
+
+    initComponent() {
+        this.autocompleteService = new window.google.maps.places.AutocompleteService();
+        this.setState({ initialized: true });
+    }
+
+    handleChange(option) {
+        const { onChange } = this.props;
+
+        if (!option) {
+            this.setState({
+                value: null,
+                selectedPlace: option,
+            });
+            onChange(option);
+            return;
+        }
+
+        const { places, suggestions } = this.state;
+        const value = places.length > 0 ? places[suggestions.indexOf(option)] : option;
+
+        this.setState({
+            value,
+            selectedPlace: {
+                label: option.label,
+                description: option.description,
+                icon: <SelectedLocationIcon />,
+            },
+        });
+
+        onChange(value);
+    }
+
+    handleSearch(value) {
+        const { initialized } = this.state;
+        if (!initialized) return;
+
+        if (value && this.state.searchValue !== value) {
+            this.setState({ searchValue: value });
```

Comment:
I think this state is not needed, since this the comparation:
```this.state.searchValue !== value``` is not needed, always you get here this values will be differents and pass value to processSearchResults and handleError as argument and then youn can remove it from state.